### PR TITLE
refector:  replace external package "pointer" to "kubevirt/pkg/pointer" 

### PR DESCRIPTION
**Pull request by yj-yoo**
### What this PR does

The external pointer package has been deprecated, so change the package to "pkg/pointer".
(This PR addresses issue #11316  and is a further refinement of PR #11342.)
reference: https://pkg.go.dev/k8s.io/utils/pointer

Fixes 

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [x] PR: The PR description is expressive enough and will help future contributors
- [x] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [x] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

**Review by EdDev — COMMENTED** [review]
I prefer the simple `pkg/pointer` we have.
There is no reason to depend on an external package for such a simple helper.

**Comment by yj-yoo** [comment]
@EdDev 
Oh... I found the package, and I agree with you. For that reason, both "/k8s.io/utils/pointer" and "/k8s.io/utils/ptr" depend on external packages, so I should replace them with "pkg/pointer" as you mentioned.

File: pkg/instancetype/instancetype_test.go
Comment (EdDev) [inline pkg/instancetype/instancetype_test.go:25]: What are these?

File: pkg/virt-api/rest/subresource_test.go
Comment (EdDev) [inline pkg/virt-api/rest/subresource_test.go:44]: This should appear in a different group. Please add a newline above.

File: pkg/virt-operator/resource/generate/components/crds.go
Comment (EdDev) [inline pkg/virt-operator/resource/generate/components/crds.go:40]: You should keep the newline here, these are two different groups.

**Comment by yj-yoo** [comment]
@EdDev Thank you😊 I pushed again.

**Review by EdDev — COMMENTED** [review]
Thank you.
I had some minor comments inline.
It would also be nice if you could change the aliases used so everything will be called `pointer.P`, but it is a nice-to-have.

I just hope you will get an approve soon, as reviewing this take some time and the chances you will collide with other PRs that get merged before you are high .
Take into account that both reviewers/approvers and merge collisions are a problem with big PRs.

Lastly, please make sure to squash all commits at the end. Only if each commit has a logical purpose keep it in the history.

**Comment by EdDev** [comment]
> Release note

Please add there `NONE` to resolve the `do-not-merge/release-note-label-needed` label.

File: pkg/instancetype/instancetype_test.go
Comment (yj-yoo) [reply pkg/instancetype/instancetype_test.go:25]: This is not my change. During the make generate command process, the line was changed and output as a change.

File: pkg/virt-operator/resource/generate/components/crds.go
Comment (yj-yoo) [reply pkg/virt-operator/resource/generate/components/crds.go:40]: This is the deleted line. The line added on the right is separated by a newline above and below. Should I change the position?

File: pkg/instancetype/instancetype_test.go
Comment (EdDev) [reply pkg/instancetype/instancetype_test.go:25]: Ohh sorry, now I see it just moved.

Comment (yj-yoo) [reply pkg/instancetype/instancetype_test.go:25]: It's okay, don't worry.😁 "make generate" and IDE are often misunderstood.

File: pkg/virt-operator/resource/generate/components/crds.go
Comment (akalenyu) [reply pkg/virt-operator/resource/generate/components/crds.go:40]: I think we can enforce these things via `goimports`:
https://github.com/kubevirt/containerized-data-importer/pull/3097

Comment (EdDev) [reply pkg/virt-operator/resource/generate/components/crds.go:40]: I think we use it already through bazel. It does not work well enough, especially between different 3rd party domains. And there is a need to configure the "self" domain as well.

**Comment by yj-yoo** [comment]
@AlonaKaplan 
okay. I checked, modified it again, and pushed it.

**Comment by EdDev** [comment]
Please squash the fix into the relevant commit so the history will be clean.

> Merge branch 'main' into pointer-to-ptr

It needs to be a rebase over main, not a merge.

**Comment by yj-yoo** [comment]
@EdDev 
As always, thank you for your advice.

**Comment by vladikr** [comment]
/approve

It looks good to me. Sorry to see the `pointer.To` go away but that's fine ;)

**Review by xpivarc — COMMENTED** [review]
/approve

**Comment by EdDev** [comment]
For some reason, the `pull-kubevirt-verify-go-mod` fails consistently.

**Comment by oshoval** [comment]
Doesn't this https://github.com/kubevirt/kubevirt/compare/69e1b3a36c9ef9670fef9ba69ed911ded685e94f..2c39fd6266cd9685b81599f4e391198386f6251a can mask some problem ?
if the test still fails, but without panic it is fine of course (until the root cause is fixed)

**Comment by yj-yoo** [comment]
@oshoval 
i get it.
It's embarrassing because it's a simple function change, but I'll try to find the root cause. thanks.

**Comment by EdDev** [comment]
> /test pull-kubevirt-verify-go-mod

@brianmcarey , [this](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/11348/pull-kubevirt-verify-go-mod/1763425708253646848) keeps happening and at this hour I would not expect load.

/test pull-kubevirt-verify-go-mod

**Comment by brianmcarey** [comment]
> > /test pull-kubevirt-verify-go-mod
> 
> @brianmcarey , [this](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/11348/pull-kubevirt-verify-go-mod/1763425708253646848) keeps happening and at this hour I would not expect load.
> 
> /test pull-kubevirt-verify-go-mod

This looks like an actual issue. 

Can you try running `make deps-sync` locally against this PR to see if it completes successfully?

**Comment by yj-yoo** [comment]
> > > /test pull-kubevirt-verify-go-mod
> > 
> > 
> > @brianmcarey , [this](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/11348/pull-kubevirt-verify-go-mod/1763425708253646848) keeps happening and at this hour I would not expect load.
> > /test pull-kubevirt-verify-go-mod
> 
> This looks like an actual issue.
> 
> Can you try running `make deps-sync` locally against this PR to see if it completes successfully?

I tried this command locally yesterday and got the following results:
```
go: github.com/openshift/client-go@v0.0.0: reading github.com/openshift/client-go/go.mod at revision v0.0.0: unknown revision v0.0.0
```
Should I research this error?

**Comment by EdDev** [comment]
@yj-yoo , the problem is with these changes:
```
staging/src/kubevirt.io/client-go/api/BUILD.bazel
staging/src/kubevirt.io/client-go/api/defaults_test.go
staging/src/kubevirt.io/client-go/api/schema_test.go
```

Please revert the changes on these files and send again.
These will have to be solved differently, as depending on the kubevirt project from this staging thing is problematic.

**Comment by yj-yoo** [comment]
> @yj-yoo , the problem is with these changes:
> 
> ```
> staging/src/kubevirt.io/client-go/api/BUILD.bazel
> staging/src/kubevirt.io/client-go/api/defaults_test.go
> staging/src/kubevirt.io/client-go/api/schema_test.go
> ```
> 
> Please revert the changes on these files and send again. These will have to be solved differently, as depending on the kubevirt project from this staging thing is problematic.

oh! Thank you. This was the part I was most suspicious of, so I tried to roll back and test it first. I'm late for work at work. Thank you again for saving me the trouble.

**Review by EdDev — APPROVED** [review]
Thank you!

**Comment by EdDev** [comment]
> Oh sorry. It was just a rebase, but was it an unnecessary action? Or should I comment that it is a “rebase”?

Commenting that you rebased will help a lot, thanks.
Rebasing is only needed if the label appears on the PR, otherwise prow has no merge conflicts and can do it itself.

**Comment by yj-yoo** [comment]
> > Oh sorry. It was just a rebase, but was it an unnecessary action? Or should I comment that it is a “rebase”?
> 
> Commenting that you rebased will help a lot, thanks. Rebasing is only needed if the label appears on the PR, otherwise prow has no merge conflicts and can do it itself.

Oh, I understand. Thank you, I will work carefully!

**Comment by oshoval** [comment]
Btw, there is usually no need to rebase, unless something breaks in run time / compilation
Prow rebase each and every lane before it runs, 
and before merging Tide makes sure to test it on latest main

Thanks

**Review by EdDev — APPROVED** [review]

**Pull request closed without merging**
